services/code.service.js

```diff
@@ -380,6 +386,93 @@ const _getAiScore = async (langConfig, question, response, points, userAnswer, r
     }
 }
 
+const _executeStatement = (db, sql) => {
+    return new Promise((resolve, reject) => {
+        db.all(sql, function(err, rows) {
+            if (err) {
+                reject(err);
+            } else {
+                resolve(rows)
+            }
+        })
+    })
+}
+
+const _executeSqlQueries = async (dbPath, queries) => {
+    const db = new sqlite3.Database(dbPath, sqlite3.OPEN_READWRITE, (err) => {
+        if (err) {
+            throw new Error(
+                'Was not able to connect to the SQLite database: ' + err.message,
+            )
+        }
+    })
+
+    const cleanedQueries = []
+    try {
+        const ast = parser(queries);
+        if (!ast) {
+            return { data: [] }
+        }
+        for (const statement of ast.statement) {
+            cleanedQueries.push(generate(statement))
+        }
+    } catch (err) {
+        return { error: true, data: err.message }
+    }
+
+    for (let i = 0; i < cleanedQueries.length; i++) {
+        try {
+            const res = await _executeStatement(db, cleanedQueries[i])
+            if (i == cleanedQueries.length - 1) {
+                return { data: res }
+            }
+        } catch (err) {
+            logger.error(err)
+            return {
+                error: true, data: `${err.message} at statement ${i + 1}`
+            }
+        }
+    }
+}
+
+const _downloadSqliteDatabase = async (fileUrl, dbPath) => {
+    const writer = fs.createWriteStream(dbPath)
+    const response = await axios({
+        url: fileUrl,
+        method: 'GET',
+        responseType: 'stream',
+    })
+
+    response.data.pipe(writer)
+
+    return new Promise((resolve, reject) => {
+        writer.on('finish', resolve)
+        writer.on('error', reject)
+    })
+}
+
+const _executeSqlite3Query = async (req, res, response) => {
+    const dbPath = '/tmp/database.db'
+    try {
+        const dbDirectory = path.dirname(dbPath)
+        if (!fs.existsSync(dbDirectory)) {
+            fs.mkdirSync(dbDirectory, { recursive: true })
+        }
+        if (!fs.existsSync(dbPath)) {
+            fs.closeSync(fs.openSync(dbPath, 'w'))
+        }
+        await _downloadSqliteDatabase(req.stdin, dbPath)
+        const queryResults = await _executeSqlQueries(dbPath, req.script, response)
+        if (queryResults.error) {
+            response.error = 1
+        }
+        response.output = JSON.stringify(queryResults.data)
+    } catch (err) {
```

Comment:
I don't see a cleanup of the file `/tmp/database.db` once the request is served, please check if that is needed.